cmd/kube-proxy/app/server.go

```diff
@@ -384,17 +385,26 @@ type ProxyServer struct {
 // createClients creates a kube client and an event client from the given config and masterOverride.
 // TODO remove masterOverride when CLI flags are removed.
 func createClients(config componentconfig.ClientConnectionConfiguration, masterOverride string) (clientset.Interface, v1core.EventsGetter, error) {
+	var kubeConfig *rest.Config
+	var err error
+
+	// If both kubeconfig and master URL are empty, use service account.
 	if len(config.KubeConfigFile) == 0 && len(masterOverride) == 0 {
-		glog.Warningf("Neither --kubeconfig nor --master was specified. Using default API client. This might not work.")
-	}
+		glog.Info("Neither --kubeconfig nor --master was specified. Using service account.")
+		kubeConfig, err = rest.InClusterConfig()
```

Comment:
Yes there is a chicken-and-egg problem inherent in implementing service proxy with in-cluster config. I ran into same issue when implementing a service proxy. Actual master IP need to be accessed to read the service and endpoints info to set up `kubernetes` service.